core/src/main/scala/org/apache/spark/TestUtils.scala

```diff
@@ -168,6 +168,27 @@ private[spark] object TestUtils {
     createCompiledClass(className, destDir, sourceFile, classpathUrls)
   }
 
+  /** Create a dummy compile jar for a given package, classname.  Jar will be placed in destDir */
+  def createDummyJar(destDir: String, packageName: String, className: String): String = {
+    val srcDir = new File(destDir, packageName)
+    srcDir.mkdirs()
+    val excSource = new JavaSourceFromString(new File(srcDir, className).getAbsolutePath,
```

Comment:
It's fine to push commit and use AppVeyor IMHO. There is a guide for this - https://github.com/apache/spark/blob/master/R/WINDOWS.md if you are willing to manually test but it is kind of a grunt work. In my case, I use my personal AppVeyor account.

Otherwise, I can send a PR for your branch if you are happy with waiting till this weekend :).